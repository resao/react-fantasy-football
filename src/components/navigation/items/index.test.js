import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import NavigationItems from "./";
import NavigationItem from "./item/";

configure({ adapter: new Adapter() });

describe("<NavigationItems/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavigationItems />);
  });

  it("should render correct <NavigationItem/> elements if not authenticated", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct <NavigationItem/> elements if authenticated", () => {
    wrapper.setProps({ isAuthenticated: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should include a logout link if authenticated", () => {
    wrapper.setProps({ isAuthenticated: true });
    expect(
      wrapper.contains(<NavigationItem link="/logout">Logout</NavigationItem>)
    ).toEqual(true);
  });
});
