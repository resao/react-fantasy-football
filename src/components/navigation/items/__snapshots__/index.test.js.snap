// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems/> should render correct <NavigationItem/> elements if authenticated 1`] = `
<ul
  className="items"
>
  <_default
    exact={true}
    link="/"
  >
    Team Builder
  </_default>
  <_default
    link="/saves"
  >
    Saves
  </_default>
  <_default
    link="/logout"
  >
    Logout
  </_default>
</ul>
`;

exports[`<NavigationItems/> should render correct <NavigationItem/> elements if not authenticated 1`] = `
<ul
  className="items"
>
  <_default
    exact={true}
    link="/"
  >
    Team Builder
  </_default>
  <_default
    link="/auth"
  >
    Authenticate
  </_default>
</ul>
`;
